types/src/crypto.rs

```diff
@@ -33,3 +43,53 @@ pub fn blake2b<T: AsRef<[u8]>>(data: T) -> [u8; BLAKE2B_DIGEST_LENGTH] {
     });
     result
 }
+
+/// HashAlgoType
+#[repr(u8)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "datasize", derive(DataSize))]
+#[cfg_attr(feature = "json-schema", derive(JsonSchema))]
+pub enum HashAlgoType {
+    /// Blake2b
+    Blake2b = 0,
+    /// Blake3
+    Blake3 = 1,
+}
+
+impl ToBytes for HashAlgoType {
```

Comment:
You can skip ToBytes/FromBytes for this type as it's easier to pass it by value. Same with `impl AsRef` I think.

Reply:
I removed both, and introduced `TryFrom<u8>` - see https://github.com/casper-network/casper-node/pull/4412/commits/2c70942af76e7e46b5c9d1fa74445d8d5131e555.

---

Btw. handling error was crazy, as compiler could not determine that `casper_types::bytesrepr::Error` should be converted into `casper_execution_engine::core::execution::error`, and then into `casper_wasmi_core::trap::Trap`. Finally I found working line:
```rs
let hash_algo_type = HashAlgoType::try_from(hash_algo_type as u8).map_err(|e| Error::from(e))?;
```